src/Symfony/Component/Uid/UuidFactory.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uid;
+
+class UuidFactory
+{
+    private $defaultNamedVersion;
+    private $defaultTimedVersion;
+
+    /**
+     * @var Uuid|null
+     */
+    private $defaultNamespace;
+
+    /**
+     * @var Uuid|null
+     */
+    private $defaultNode;
+
+    /**
+     * @param string|null $defaultNamedVersion 5 or 3
+     * @param string|null $defaultTimedVersion 6 or 1
+     */
+    public function __construct(string $defaultNamedVersion = null, string $defaultTimedVersion = null)
```

Comment:
just for my understanding, why didn't you use `int`?

Reply:
Because I cascaded the type from the config but since the type constants are int, I guess we should be coherent and use int here too? Should I also switch to integer nodes in the config?

Reply:
Mmm, I'm not sure this makes sense. The config could accept int nodes also.

Reply:
I changed everything to int.